### PR TITLE
Add the iotdb-cluster.properties configuration file and start stop cleanup script

### PR DESCRIPTION
**Pull request by 2b3c511**
Add iotdb-cluster.properties, start-all.sh, stop-all.sh, clean-all.sh, clean-all.bat, clean-datanode.sh, clean-datanode.bat, and clea n-confignode.sh and clean-confignode.bat files

### CODE REVIEW

File: iotdb-core/node-commons/src/assembly/resources/conf/iotdb-cluster.properties [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+# specific language governing permissions and limitations
+# under the License.
+#
+
+
+# this function is for parsing the variables like "A=B" in  `start-server.sh -D A=B`
+# The command just parse IOTDB-prefixed variables and ignore all other variables
+
+# Configure ConfigNodes machine addresses separated by ,
+confignode_address_list=127.0.0.1
```
Comment (CritasWang) [inline iotdb-core/node-commons/src/assembly/resources/conf/iotdb-cluster.properties:25]: Default comment out

File: iotdb-core/node-commons/src/assembly/resources/sbin/start-all.sh [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+#
+
+if [ -z "${IOTDB_HOME}" ]; then
+    export IOTDB_HOME="`dirname "$0"`/.."
+fi
+# iotdb-cluster.properties does not exist, the current 1C1D is started
+if [ ! -f ${IOTDB_HOME}/conf/iotdb-cluster.properties ]; then
+  exec ${IOTDB_HOME}/sbin/start-standalone.sh
+else
+  confignodeStr=$(sed '/^confignode_address_list=/!d;s/.*=//' "${IOTDB_HOME}"/conf/iotdb-cluster.properties)
```
Comment (CritasWang) [inline iotdb-core/node-commons/src/assembly/resources/sbin/start-all.sh:28]: This repeated paragraph should be extracted as a common script

**Review by CritasWang — CHANGES_REQUESTED** [review]

**Comment by ppppoooo** [comment]
[

](url)

File: iotdb-core/node-commons/src/assembly/resources/sbin/clean-confignode.bat [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+if "%cn_consensus_dir%"=="" (
+set "cn_consensus_dir=data\confignode\consensus"
+)
+setlocal enabledelayedexpansion
+set "cn_consensus_dir=!cn_consensus_dir:%delimiter%= !"
+for %%i in (%cn_consensus_dir%) do (
+  set "var=%%i"
+    if "!var:~0,2!"=="\\" (
+        rmdir /s /q "%%i" 2>nul
+    ) else if "!var:~1,3!"==":\\" (
```
Comment (ppppoooo) [inline iotdb-core/node-commons/src/assembly/resources/sbin/clean-confignode.bat:73]: Is this position always false

File: iotdb-core/node-commons/src/assembly/resources/sbin/clean-all.sh [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+for datanodeIP in ${unique_array[@]};do
+  hasConfigNode="false"
+  for ((i=0; i<${#confignodeIps[@]}; i++))
+      do
+          # 检查元素是否包含指定字符
+          if [[ "${confignodeIps[$i]}" == *"$datanodeIP"* ]]; then
+              # 打印包含指定字符的元素
+              hasConfigNode="true"
+              # 从数组中删除这个元素
+              unset 'confignodeIps[$i]'
```
Comment (ppppoooo) [inline iotdb-core/node-commons/src/assembly/resources/sbin/clean-all.sh:74]: Doesn't deleting while traversing overrun the array?

File: iotdb-core/node-commons/src/assembly/resources/sbin/clean-all.sh [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+  account=$(sed '/^ssh_account=/!d;s/.*=//' "${IOTDB_CLUSTER_PATH}")
+#  echo $confignodeIps $datanodeIps $confignodePaths $datanodePaths $account $serverPort
+fi
+
+function validateParam() {
+  if [[ -z $1 || -z $2 ||  -z $3 ||  -z $4 ||  -z $5 ||  -z $6 ]]; then
+    echo "The iotdb-cluster.properties file is incomplete, the current 1C1D will be cleaned ... "
+    exec ${IOTDB_HOME}/sbin/clean-datanode.sh -f > /dev/null 2>&1 &
+    exec ${IOTDB_HOME}/sbin/clean-confignode.sh -f> /dev/null 2>&1 &
+    exit
```
Comment (ppppoooo) [inline iotdb-core/node-commons/src/assembly/resources/sbin/clean-all.sh:51]: Why isn't data deleted from this location?

File: iotdb-core/node-commons/src/assembly/resources/sbin/clean-datanode.bat [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+)
+
+for /f  "eol=# tokens=2 delims==" %%i in ('findstr /i "^dn_consensus_dir"
+  %IOTDB_DATANODE_CONFIG%') do (
+  set dn_consensus_dir=%%i
+)
+if "%dn_consensus_dir%"=="" (
+    set "dn_consensus_dir=data\\datanode\\consensus"
+)
+setlocal enabledelayedexpansion
```
Comment (ppppoooo) [inline iotdb-core/node-commons/src/assembly/resources/sbin/clean-datanode.bat:87]: Is it enough to set it up once

**Comment by 2b3c511** [comment]
> 
> 
>

File: iotdb-core/node-commons/src/assembly/resources/sbin/clean-all.sh (hunk 5 above)
Comment (2b3c511) [reply iotdb-core/node-commons/src/assembly/resources/sbin/clean-all.sh:53]: done

File: iotdb-core/node-commons/src/assembly/resources/sbin/clean-datanode.bat (hunk 6 above)
Comment (2b3c511) [reply iotdb-core/node-commons/src/assembly/resources/sbin/clean-datanode.bat]: done

File: iotdb-core/node-commons/src/assembly/resources/sbin/clean-all.bat [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+SET /P CLEAN_SERVICE="Do you want to clean all the data in the IoTDB ? y/n (default n): "
+IF NOT "%CLEAN_SERVICE%"=="y" IF NOT "%CLEAN_SERVICE%"=="Y" (
+  ECHO Exiting...
+  EXIT /b
+)
+
+pushd %~dp0..
+if NOT DEFINED IOTDB_HOME set IOTDB_HOME=%cd%
+popd
```
[the remark this answers is not in the document]
Comment (2b3c511) [reply iotdb-core/node-commons/src/assembly/resources/sbin/clean-all.bat:30]: pushd %~dp0..   It pushes the current directory onto the stack and switches to the directory above the current directory.If IOTDB_HOME is not defined, set it to the current working directory path.The purpose of popd is to return to the previously pushed directory, that is, to the original working directory

**Review by CritasWang — APPROVED** [review]

**Pull request merged**